ui/list-item-menu.reel/list-item-menu.js

```diff
@@ -0,0 +1,1162 @@
+                            // should close a side if the minimum distance has not been reached.
+                            this._shouldClose = true;
+                        }
+                    }
+                }
+
+                this._resetTranslateContext();
+            } else {
+                this._closeIfNeeded();
+            }
+        }
+    },
+
+    handleTranslateCancel: {
+        value: function () {
+            this._resetTranslateContext();
+            this._isTranslating = false;
+            this._direction = null;
+        }
+    },
+
+    handlePressStart: {
+        value: function (event) {
+            var target = event.targetElement;
+
+            if (this.element !== target && !this.element.contains(target)) {
+                this.close();
+            }
+        }
+    },
+
+    handlePress: {
+        value: function () {
+            this._closeIfNeeded();
+        }
+    },
+
+    _addDragEventListeners: {
+        value: function () {
+            this._translateComposer.addEventListener('translate', this);
+            this._translateComposer.addEventListener('translateCancel', this);
+        }
+    },
+
+    _removeDragEventListeners: {
+        value: function () {
+            this._translateComposer.removeEventListener('translate', this);
+            this._translateComposer.removeEventListener('translateCancel', this);
+        }
+    },
+
+    _resetTranslateContext: {
+        value: function () {
+            this._removeDragEventListeners();
+            this._startTimestamp = 0;
+            this._distance = null;
+            this._hasReachEnd = false;
+            this.needsDraw = true;
+        }
+    },
+
+    /**  
+     * 
+     *  Draw cycle management 
+     * 
+     */
+
+    willDraw: {
+        value: function () {
+            if (
+                !this._dragElementRect ||
+                this._dragElementRect.width === 0 ||
+                this._forceComputingBoundaries
+            ) {
+                this._dragElementRect = this.dragElement.getBoundingClientRect();
+                this._hotCornersElementRect = this.hotCornersElement.getBoundingClientRect();
+                this._leftButtons = this.leftOptionsElement.querySelectorAll('button');
+                this._rightButtons = this.rightOptionsElement.querySelectorAll('button');
+
+                if ((this._rightButtons && this._rightButtons.length > 3) ||
+                    (this._leftButtons && this._leftButtons.length > 3)
+                ) {
+                    throw new Error(
+                        'the list item menu component doesn\'t support' +
+                        'more than 3 buttons per slidding side'
+                    );
+                }
+
+                var hasLeftButtons = this._leftButtons && this._leftButtons.length > 0,
+                    hasRightButtons = this._rightButtons && this._rightButtons.length > 0;
+
+                this._updateButtonPositions();
```

Comment:
This is modifying the DOM in wilDraw, could it be moved at the beginning of draw instead?